### PR TITLE
Feature: idrac_storage_volume

### LINKED ISSUE #635: [BUG]: dellemc_idrac_storage_volume - No such controller found! when trying to create RAID 1 with NVMe disks
### Bug Description

hi 
got brand new R650 with 2 NVMe disks and getting error No such controller found! when trying to create RAID 1

### Component or Module Name

dellemc_idrac_storage_volume

### Ansible Version

Ansible 2.16.3

### Python Version

Python 3.11.7

### iDRAC/OME/OME-M version

IDRAC 7.10.05.00 

### Operating System

CentOS 8

### Playbook Used

  ```
 - name: Create RAID 1 on NVMe
      dellemc.openmanage.dellemc_idrac_storage_volume:
        idrac_ip: "{{ lom_ip }}"
        idrac_user: "{{ lom_username }}"
        idrac_password: "{{ lom_password }}"
        state: create
        controller_id: BOSS.Integrated.1-1
        volume_type: RAID 1
        span_depth: 1
        span_length: 2
        raid_init_operation: Fast
        validate_certs: false
        volumes:
          - name: VDISK1
            drives:
              id:
                - Disk.Direct.0-0:BOSS.Integrated.1-1
                - Disk.Direct.1-1:BOSS.Integrated.1-1
```

### Logs

```
{
    "msg": "No such controller found!",
    "invocation": {
        "module_args": {
            "idrac_ip": "10.10.10.10",
            "idrac_user": "automation",
            "idrac_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "create",
            "controller_id": "BOSS.Integrated.1-1",
            "volume_type": "RAID 1",
            "span_depth": 1,
            "span_length": 2,
            "raid_init_operation": "Fast",
            "validate_certs": false,
            "volumes": [
                {
                    "name": "VDISK1",
                    "drives": {
                        "id": [
                            "Disk.Direct.0-0:BOSS.Integrated.1-1",
                            "Disk.Direct.1-1:BOSS.Integrated.1-1"
                        ]
                    }
                }
            ],
            "number_dedicated_hot_spare": 0,
            "disk_cache_policy": "Default",
            "write_cache_policy": "WriteThrough",
            "read_cache_policy": "NoReadAhead",
            "stripe_size": 65536,
            "raid_reset_config": "False",
            "idrac_port": 443,
            "timeout": 30,
            "volume_id": null,
            "capacity": null,
            "media_type": null,
            "protocol": null,
            "ca_path": null
        }
    },
    "_ansible_no_log": false,
    "changed": false
}

```
### Steps to Reproduce

run the playbook on the server with NVMe disks

### Expected Behavior

RAID 1 created

### Actual Behavior

error

### Screenshots

_No response_

### Additional Information
```

{
    "ansible_facts": {
        "alv_system_inventory": {
            "PCIeSSDDisk": [
                {
                    "FQDD": "Disk.Direct.0-0:BOSS.Integrated.1-1",
                    "RemainingRatedWriteEndurance": "100",
                    "Model": "DELL NVME ISE PE8010 RI M.2 480GB",
                    "Size": "447.13",
                    "BusProtocol": "PCIe",
                    "DeviceProtocol": "NVMe 1.3",
                    "MediaType": "SSD",
                    "Manufacturer": "SK hynix",
                    "ProductID": "DELL NVME ISE PE8010 RI M.2 480GB",
                    "SerialNumber": "NDC9N4975I1509H10",
                    "DriveFormFactor": "5",
                    "MaximumCapableSpeed": "8 GT/s",
                    "Revision": "1.2.0",
                    "PrimaryStatus": "Healthy",
                    "PredictiveFailureState": "Healthy",
                    "Key": "Disk.Direct.0-0:BOSS.Integrated.1-1"
                },
                {
                    "FQDD": "Disk.Direct.1-1:BOSS.Integrated.1-1",
                    "RemainingRatedWriteEndurance": "100",
                    "Model": "DELL NVME ISE PE8010 RI M.2 480GB",
                    "Size": "447.13",
                    "BusProtocol": "PCIe",
                    "DeviceProtocol": "NVMe 1.3",
                    "MediaType": "SSD",
                    "Manufacturer": "SK hynix",
                    "ProductID": "DELL NVME ISE PE8010 RI M.2 480GB",
                    "SerialNumber": "NDC9N4992I0209I1E",
                    "DriveFormFactor": "5",
                    "MaximumCapableSpeed": "8 GT/s",
                    "Revision": "1.2.0",
                    "PrimaryStatus": "Healthy",
                    "PredictiveFailureState": "Healthy",
                    "Key": "Disk.Direct.1-1:BOSS.Integrated.1-1"
                }
            ],
            "PhysicalDisk": [
                {
                    "FQDD": "Disk.Direct.0-0:BOSS.Integrated.1-1",
                    "RemainingRatedWriteEndurance": "100",
                    "FreeSize": "447.13",
                    "RaidStatus": "Ready",
                    "T10PICapability": "Not Available",
                    "DeviceDescription": "Disk 0 on Integrated BOSS 1",
                    "Model": "DELL NVME ISE PE8010 RI M.2 480GB",
                    "SerialNumber": "NDC9N4975I1509H10",
                    "Size": "447.13",
                    "UsedSize": "Not Available",
                    "MediaType": "SSD",
                    "Revision": "1.2.0",
                    "BlockSize": "Not Available",
                    "DriveFormFactor": "5",
                    "HotSpareStatus": "No",
                    "Manufacturer": "SK hynix",
                    "ManufacturingDay": "Not Available",
                    "ManufacturingWeek": "Not Available",
                    "ManufacturingYear": "Not Available",
                    "MaxCapableSpeed": "Not Available",
                    "PPID": "Not Available",
                    "SASAddress": "Not Available",
                    "Slot": "0",
                    "SupportedEncryptionTypes": "Not Available",
                    "BusProtocol": "PCIe",
                    "SecurityState": "Not Available",
                    "RAIDNegotiatedSpeed": "8 GT/s",
                    "PrimaryStatus": "Healthy",
                    "PredictiveFailureState": "Healthy",
                    "Key": "Disk.Direct.0-0:BOSS.Integrated.1-1"
                },
                {
                    "FQDD": "Disk.Direct.1-1:BOSS.Integrated.1-1",
                    "RemainingRatedWriteEndurance": "100",
                    "FreeSize": "447.13",
                    "RaidStatus": "Ready",
                    "T10PICapability": "Not Available",
                    "DeviceDescription": "Disk 1 on Integrated BOSS 1",
                    "Model": "DELL NVME ISE PE8010 RI M.2 480GB",
                    "SerialNumber": "NDC9N4992I0209I1E",
                    "Size": "447.13",
                    "UsedSize": "Not Available",
                    "MediaType": "SSD",
                    "Revision": "1.2.0",
                    "BlockSize": "Not Available",
                    "DriveFormFactor": "5",
                    "HotSpareStatus": "No",
                    "Manufacturer": "SK hynix",
                    "ManufacturingDay": "Not Available",
                    "ManufacturingWeek": "Not Available",
                    "ManufacturingYear": "Not Available",
                    "MaxCapableSpeed": "Not Available",
                    "PPID": "Not Available",
                    "SASAddress": "Not Available",
                    "Slot": "1",
                    "SupportedEncryptionTypes": "Not Available",
                    "BusProtocol": "PCIe",
                    "SecurityState": "Not Available",
                    "RAIDNegotiatedSpeed": "8 GT/s",
                    "PrimaryStatus": "Healthy",
                    "PredictiveFailureState": "Healthy",
                    "Key": "Disk.Direct.1-1:BOSS.Integrated.1-1"
                }
            ],
            "ControllerSensor": [
                {
                    "FQDD": "AHCI.Embedded.2-1",
                    "Key": "AHCI.Embedded.2-1"
                },
                {
                    "FQDD": "AHCI.Embedded.1-1",
                    "Key": "AHCI.Embedded.1-1"
                },
                {
                    "FQDD": "BOSS.Integrated.1-1",
                    "Key": "BOSS.Integrated.1-1"
                }
            ],
            "Controller": [
                {
                    "FQDD": "AHCI.Embedded.2-1",
                    "EncryptionMode": "None",
                    "SecurityStatus": "Encryption Not Capable",
                    "RollupStatus": "Unknown",
                    "Bus": "0",
                    "CacheSize": "0.0 MB",
                    "CachecadeCapability": "Cachecade Virtual Disk not supported",
                    "ControllerFirmwareVersion": "Not Available",
                    "DeviceCardDataBusWidth": "Unknown",
                    "DeviceCardManufacturer": "DELL",
                    "DeviceCardSlotLength": "2",
                    "DeviceCardSlotType": "Unknown",
                    "DeviceDescription": "Embedded AHCI 2",
                    "DriverVersion": "Not Available",
                    "EncryptionCapability": "None",
                    "MaxAvailablePCILinkSpeed": "Not Available",
                    "MaxPossiblePCILinkSpeed": "Not Available",
                    "PCISlot": "Not Available",
                    "PCIVendorID": "8086",
                    "ProductName": "Sapphire Rapids SATA AHCI Controller",
                    "SASAddress": "0",
                    "SlicedVDCapability": "Sliced Virtual Disk creation not supported",
                    "SupportControllerBootMode": "0",
                    "SupportEnhancedAutoForeignImport": "0",
                    "SupportRAID10UnevenSpans": "0",
                    "T10PICapability": "0",
                    "PrimaryStatus": "Unknown",
                    "Key": "AHCI.Embedded.2-1"
                },
                {
                    "FQDD": "AHCI.Embedded.1-1",
                    "EncryptionMode": "None",
                    "SecurityStatus": "Encryption Not Capable",
                    "RollupStatus": "Unknown",
                    "Bus": "0",
                    "CacheSize": "0.0 MB",
                    "CachecadeCapability": "Cachecade Virtual Disk not supported",
                    "ControllerFirmwareVersion": "Not Available",
                    "DeviceCardDataBusWidth": "Unknown",
                    "DeviceCardManufacturer": "DELL",
                    "DeviceCardSlotLength": "2",
                    "DeviceCardSlotType": "Unknown",
                    "DeviceDescription": "Embedded AHCI 1",
                    "DriverVersion": "Not Available",
                    "EncryptionCapability": "None",
                    "MaxAvailablePCILinkSpeed": "Not Available",
                    "MaxPossiblePCILinkSpeed": "Not Available",
                    "PCISlot": "Not Available",
                    "PCIVendorID": "8086",
                    "ProductName": "Sapphire Rapids SATA AHCI Controller",
                    "SASAddress": "0",
                    "SlicedVDCapability": "Sliced Virtual Disk creation not supported",
                    "SupportControllerBootMode": "0",
                    "SupportEnhancedAutoForeignImport": "0",
                    "SupportRAID10UnevenSpans": "0",
                    "T10PICapability": "0",
                    "PrimaryStatus": "Unknown",
                    "Key": "AHCI.Embedded.1-1"
                },
                {
                    "FQDD": "BOSS.Integrated.1-1",
                    "EncryptionMode": "6",
                    "SecurityStatus": "Encryption Not Capable",
                    "RollupStatus": "Healthy",
                    "Bus": "2A",
                    "CacheSize": "0.0 MB",
                    "CachecadeCapability": "Cachecade Virtual Disk not supported",
                    "ControllerFirmwareVersion": "2.1.13.2025",
                    "DeviceCardDataBusWidth": "4x or x4",
                    "DeviceCardManufacturer": "DELL",
                    "DeviceCardSlotLength": "1",
                    "DeviceCardSlotType": "Other",
                    "DeviceDescription": "Integrated BOSS 1",
                    "DriverVersion": "Not Available",
                    "EncryptionCapability": "None",
                    "MaxAvailablePCILinkSpeed": "Generation 3",
                    "MaxPossiblePCILinkSpeed": "Generation 3",
                    "PCISlot": "Not Available",
                    "PCIVendorID": "1B4B",
                    "ProductName": "BOSS-N1 Modular",
                    "SASAddress": "0",
                    "SlicedVDCapability": "Sliced Virtual Disk creation not supported",
                    "SupportControllerBootMode": "0",
                    "SupportEnhancedAutoForeignImport": "0",
                    "SupportRAID10UnevenSpans": "0",
                    "T10PICapability": "0",
                    "PrimaryStatus": "Healthy",
                    "Key": "BOSS.Integrated.1-1"
                }
        }
    },
    "_ansible_no_log": false,
    "changed": false
}
```

### PR DESCRIPTION
**Pull request by ABHISHEK-SINHA10**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/idrac_storage_volume.py [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+    type: int
+    default: 900
+
+requirements:
+  - "python >= 3.9.6"
+author:
+  - "Kritika Bhateja (@Kritika-Bhateja-03)"
+  - "Abhishek Sinha(@ABHISHEK-SINHA10)"
+notes:
+    - Run this module from a system that has direct access to Dell iDRAC.
```
Comment (sachin-apa) [inline plugins/modules/idrac_storage_volume.py]:
```suggestion
    - Run this module from a system that has direct access to Integrated Dell Remote Access Controller.
```

File: plugins/modules/idrac_storage_volume.py [hunk 2]
```diff
@@ -0,0 +1,825 @@
...
+      - See the examples for more details.
+  capacity:
+    type: float
+    description: Virtual disk size in GB.
+  raid_reset_config:
+    type: str
+    description:
+      - This option represents whether a reset config operation needs to be performed on the RAID controller.
+        Reset Config operation deletes all the virtual disks present on the RAID controller.
+    choices: ['True', 'False']
```
Comment (sachin-apa) [inline plugins/modules/idrac_storage_volume.py]:
```suggestion
    choices: ['true', 'false']
```

File: plugins/modules/idrac_storage_volume.py [hunk 3]
```diff
@@ -0,0 +1,825 @@
...
+    default: 'False'
```
Comment (sachin-apa) [inline plugins/modules/idrac_storage_volume.py]:
```suggestion
    default: 'false'
```

File: plugins/modules/idrac_storage_volume.py (hunk 1 above)
Comment (ABHISHEK-SINHA10) [reply plugins/modules/idrac_storage_volume.py]: Done

File: plugins/modules/idrac_storage_volume.py (hunk 3 above)
Comment (ABHISHEK-SINHA10) [reply plugins/modules/idrac_storage_volume.py]: Done

File: plugins/modules/idrac_storage_volume.py (hunk 2 above)
Comment (ABHISHEK-SINHA10) [reply plugins/modules/idrac_storage_volume.py]: Done

File: plugins/modules/idrac_storage_volume.py [hunk 4]
```diff
@@ -0,0 +1,833 @@
...
+SYSTEMS_URI = "/redfish/v1/Systems"
+iDRAC_JOB_URI = "/redfish/v1/Managers/iDRAC.Embedded.1/Jobs/{job_id}"
+CONTROLLER_NOT_EXIST_ERROR = "Specified Controller {controller_id} does not exist in the System."
+CONTROLLER_NOT_DEFINED = "Controller ID is required."
+SUCCESSFUL_OPERATION_MSG = "Successfully completed the {operation} storage volume operation."
+DRIVES_NOT_EXIST_ERROR = "No Drive(s) are attached to the specified Controller Id: {controller_id}."
+DRIVES_NOT_MATCHED = "Following Drive(s) {specified_drives} are not attached to the specified Controller Id: {controller_id}."
+NEGATIVE_OR_ZERO_MSG = "The value for the `{parameter}` parameter cannot be negative or zero."
+NEGATIVE_MSG = "The value for the `{parameter}` parameter cannot be negative."
+INVALID_VALUE_MSG = "The value for the `{parameter}` parameter are invalid."
```
Comment (rajshekarp87) [inline plugins/modules/idrac_storage_volume.py]:
```suggestion
INVALID_VALUE_MSG = "The value for the `{parameter}` parameter is invalid."
```

File: plugins/modules/idrac_storage_volume.py [hunk 5]
```diff
@@ -0,0 +1,948 @@
...
+  job_wait_timeout:
+    description:
+      - This parameter is the maximum wait time of I(job_wait) in seconds.
+      - This option is applicable when I(job_wait) is C(true).
+    type: int
+    default: 900
+
+requirements:
+  - "python >= 3.9.6"
+author:
```
Comment (rajshekarp87) [inline plugins/modules/idrac_storage_volume.py:134]: Include "Felix Stephen (@felixs88)"

File: plugins/modules/idrac_storage_volume.py (hunk 4 above)
Comment (ABHISHEK-SINHA10) [reply plugins/modules/idrac_storage_volume.py]: Done

File: plugins/modules/idrac_storage_volume.py (hunk 5 above)
Comment (ABHISHEK-SINHA10) [reply plugins/modules/idrac_storage_volume.py:134]: Done

**Review by singh-lovepreet1 — APPROVED** [review]

**Review by Kritika-Bhateja-03 — APPROVED** [review]

**Review by ShivamSh3 — APPROVED** [review]

**Review by felixs88 — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Pull request merged**